### PR TITLE
highlight problematic property in the designer when error is selected

### LINKED ISSUE #18511: highlight the field when the validation error row is selected
currently the validation error list only show the errors, in case there are many entities it could be hard to spot the problematic ones.

### PR DESCRIPTION
**Pull request by alanrenmsft**
This PR fixes #18511 

1. replace the error list implementation with the accessible solution (using list component).
2. when error item is selected, bring the problematic item into view and then highlight it (make it easier to visually spot it).
3. fix some accessibility issues.

error-navigation-light
error-navigation-dark

### CODE REVIEW

File: src/sql/workbench/browser/designer/designerPropertyPathValidator.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	 * @returns Whether the path is valid.
+	 */
+	static validate(path: DesignerPropertyPath, viewModel: DesignerViewModel): boolean {
+		// the path must have items and currently we support up to 5 items in the path.
+		if (!path || path.length === 0 || path.length > 5) {
+			return false;
+		}
+
+		for (let index = 0; index < path.length; index++) {
+			const expectingNumber = (index % 2) !== 0;
```
Comment (kburtram) [inline src/sql/workbench/browser/designer/designerPropertyPathValidator.ts:23]: the logic here isn't self-explanatory to me.  maybe a comment would help why we expect number/strings on odd/even elements?

**Review by kburtram — APPROVED** [review]
LGTM

File: src/sql/workbench/browser/designer/designerPropertyPathValidator.ts (hunk 1 above)
Comment (alanrenmsft) [reply src/sql/workbench/browser/designer/designerPropertyPathValidator.ts:35]: added comment to explain all the supported scenarios.

File: src/sql/workbench/browser/designer/designer.ts [hunk 2]
```diff
@@ -469,6 +480,74 @@ export class Designer extends Disposable implements IThemable {
...
+				if (path.length > 2) {
+					const relativePath = path.slice(2);
+					this._propertiesPane.selectProperty(relativePath);
+				}
+			}
+			this.highlightActiveElement();
+		}
+	}
+
+	private highlightActiveElement(): void {
```
Comment (abist) [inline src/sql/workbench/browser/designer/designer.ts:535]: any reason we don't just add `.highlighted` class to the elements instead and put the styling in the css? That way you wouldn't have this logic for every highlighting element elsewhere

**Review by abist — APPROVED** [review]

File: src/sql/workbench/browser/designer/designer.ts (hunk 2 above)
Comment (alanrenmsft) [reply src/sql/workbench/browser/designer/designer.ts:535]: there are only 2 properties I want to change, and both of them needs to be retrieved at runtime based on theming.

**Pull request merged**